keps/sig-api-machinery/0015-dry-run.md

```diff
@@ -150,3 +151,52 @@ A dry-run request should behave as close as possible to a regular
 request. Attempting to dry-run create an existing object will result in an
 `AlreadyExists` error to be returned. Similarly, if a dry-run update is
 performed on a non-existing object, a `NotFound` error will be returned.
+
+## kubectl
+
+For the `kubectl` client integration with server-side dry-run, we will pass
+the `dryRun` query parameter by reading the user's intent from a flag.
+
+For beta, we use `--server-dry-run` for `kubectl apply` to exercise
```

Comment:
We need at least one release of deprecation? @liggitt

Reply:
Oh @soltysh mentioned this:

> Since this was beta, you can remove this in one release, honestly (compare https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecation)

> Beta feature to EOL: 3 months or 1 release (whichever is longer)

Fixing from next release to after 1 release*

Reply:
Great, thanks!